packages/subgraph/src/plugin/pluginRepo.ts

```diff
@@ -1,15 +1,32 @@
-import {VersionCreated} from '../../generated/templates/PluginRepoTemplate/PluginRepo';
-import {PluginVersion, PluginSetup} from '../../generated/schema';
+import {
+  ReleaseMetadataUpdated,
+  VersionCreated
+} from '../../generated/templates/PluginRepoTemplate/PluginRepo';
+import {
+  PluginVersion,
+  PluginSetup,
+  PluginRelease
+} from '../../generated/schema';
 
 export function handleVersionCreated(event: VersionCreated): void {
-  let id = `${event.address.toHexString()}_${event.params.release.toString()}_${event.params.build.toString()}`;
+  let pluginRepoId = event.address.toHexString();
   let pluginSetupId = event.params.pluginSetup.toHexString();
-  let entity = new PluginVersion(id);
+  let pluginRelease = event.params.release.toString();
+
+  let pluginVersionId = pluginRepoId
+    .concat('_')
+    .concat(pluginRelease)
+    .concat('_')
+    .concat(event.params.build.toString());
+
+  let pluginReleaseId = pluginRepoId.concat('_').concat(pluginRelease);
+
+  let entity = new PluginVersion(pluginVersionId);
   entity.pluginRepo = event.address.toHexString();
-  entity.release = event.params.release;
+  entity.release = pluginReleaseId;
   entity.build = event.params.build;
   entity.pluginSetup = pluginSetupId;
-  entity.metadata = event.params.buildMetadata;
+  entity.metadata = event.params.buildMetadata.toString();
   entity.save();
 
   let pluginSetupEntity = PluginSetup.load(pluginSetupId);
```

Comment:
Let's move this on top(before creating the `PluginVersion`) just for better predictability.

Reply:
amended. thanks